src/main.js

```diff
@@ -95,19 +96,113 @@ buttons.forEach((button) => {
   });
 });
 
-//shrink header on scroll
+// Object Literal Module
+
+// (function () {
+//   const buttons = {
+//     init: function () {
+//       this.cacheDom();
+//       this.bindEvents();
+//       // this.render();
```

Comment:
Regarding your question:

> The tutorial mentions that every module would have a cacheDom(),  bindEvents()  and  render()  methods but he's using jquery and mustache so I don't get what render() is actually doing and if I need it or not.


The render is the method responsible for creating the needed HTML and mutating the DOM. In this case, to create the list of destinations (figure). So, yes, following this pattern, every time a module returns a piece of HTML it should have a render method to be more explicit that "it renders HTML". Note that this is just a convention, it is not a rule ;)

Regarding jQuery and Mustache...

I'd like you to keep doing this project without `jQuery` so that you keep practice "pure JS vanilla". The problem with using jQuery right away is that people do not understand what's jQuery vs JS vanilla. In a future project I recommend you try out jQuery so that you are aware of it, but for now, let's keep it vanilla style, okay? I guarantee you that you'll learn DOM manipulation and JS much better without it :D

In terms of Mustache, I have the same opinion as jQuery with a slight difference: When you are finished doing the module for the destinations list and the form to add a new destination, your next step can be to refactor the code to use Mustache. I'm not asking you to do that now so you can first learn how to create HTML elements and manipulate them through vanilla JS. Once you are finished with those 2 modules, feel free to add mustache! :D

You may asking: Why can't I add jQuery too after it? Good question. Well, my main reason is because Mustache is very similar to how modern UI Libraries (ex React, Vue) are created today when it comes to "template systems" and  how we create HTML through JS. On the other hand, jQuery is this overload of JS "shortcuts" that in my opinion, aren't no longer necessary with modern ES6 Javascript. Besides that and you won't need it with modern UI Libraries. I hope to be clear explaining this to you, but feel free to ask follow up questions :)

Reply:
I'm all for taking things one step at a time so I'm in no rush to move on from vanilla JS, I understand the importance of having a solid foundation. The only reason I can think of to learn some jquery is to be able to understand it in codepens/stackoverflow etc. Does keeping it vanilla also mean no plugins though? Like if I was to add a carousel, how important is it knowing how to do it by hand with so many existing options?

Reply:
>  learn some jquery is to be able to understand it in codepens/stackoverflow

Oh, that's a very good scenario to actual learn a bit of jQuery. It's totally okay to learn it, there are tones of free tutorials explaining the jQuery basics, that's more than enough to understand its basics. After it, you can just read their [API docs](https://api.jquery.com/) each time you find a new method ;)

> Does keeping it vanilla also mean no plugins though?
Not at all, of course, you can, the community is full of packages to be shared! For example, `moment` that I told you about. Another popular package is [lodash](https://lodash.com/). You can just add it as you go.

> How important is it knowing how to do it by hand with so many existing options
I'd say it's somehow important when it's common easy things, even if just out of curiosity so that you really understand how it was made. Also, if you find bugs in the future, you can fix them or even add new features. Or, create your own customized solution. For fun, I always like to check the source code and try to understand how it was done, but then I just endup using existing solutions to save time.

Reply:
Cool, I'll make some time for getting familiar with jquery basics and researching packages.

Reply:
Oh I see it, the problem has to do with the path, it's wrong.


The relative path is always relative to the current file location. The current file is at `scripts/modules`. The `destinationList` is one folder behind, so we need to write `../`. When we use `./` it means "current folder".

```diff
- import destinationsList from "./scripts/destinationList.js";
+ import destinationsList from "../destinationList.js";
```

Reply:
I'm lost 😄 the paths are fine as far as I can see, no errors but the images are still not displaying

Reply:
It seems it worked after all (as we spoke in discord, it was a refresh problem) ;)